fabric-mod-protocol-api-v1/src/main/java/net/fabricmc/fabric/impl/modprotocol/TextUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.modprotocol;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import net.minecraft.text.MutableText;
+import net.minecraft.text.Text;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+
+public class TextUtil {
+	private static final Map<String, String> FALLBACK_TRANSLATIONS = new HashMap<>();
+
+	static {
+		try {
```

Comment:
Hmm, I always thought vanilla loaded the English one as a fallback.

Reply:
It does, but only for vanilla (server.jar) one. Fabric has a patch for loading modded ones, but it might not be present when this is used standalone.

Reply:
Resource Loader is expected to be always there, I think.

Reply:
It does work like this currently (at least for clients with fabric api), but the text is created (and does setup the fallback property) in common so the same message can be displayed on vanilla clients